amiibo_functions.py

```diff
@@ -112,6 +120,8 @@ def transplant(self, bin_location, character, saveAs_location, randomize_SN=Fals
             Hit any button to close the program.'''.format(bin))
             exit()
         dump.data[84:92] = bytes.fromhex(hex_tag)
+        if character == 'Mii Brawler' or character == 'Mii Swordfighter' or character == 'Mii Gunner':
```

Comment:
The bin at bin_location should be checked to see if it is a mii fighter. If going from mii fighter to mii fighter no mii data changes are necessary. If going from mii fighter to non-mii, then all mii data should be reset to defaults (including mii moves, voice, and color).